### ISSUE TITLE
Using display hooks in order view produces error

### ISSUE THREAD

**Issue by krekken** [PrestaShop/PrestaShop#26471]
#### Describe the bug

After updating my container from 1.7.7.8 to 1.7.8, the modules i'm working on that uses some of the display hooks in order view, produces a render error. The two hooks i tried to use are: `displayAdminOrderMain` and `displayAdminOrderSide`.

Before upgrading everything worked fine, so i'm fairly sure this is a bug.

The error(i screenshot the whole profiler down below):
```
An exception has been thrown during the rendering of a template ("Expected argument of type "Symfony\Component\HttpFoundation\Request", "NULL" given").
```
#### Steps to reproduce
--------------------------------------------------------------------------------------------
> Hello @krekken & @maldarol
> 
> Could you please provide clear and concise steps to reproduce the issue pray Please check and feedback.
> 
> Thanks!

i made a module, that reproduces the problem, but i'll describe what i did, in case you don't want to rely on that. 
https://github.com/krekken/ordererrorreproduce

So i installed the hook like this, in the main php file:

```php

    public function install()
    {
        Configuration::updateValue('ORDERERRORREPRODUCE_LIVE_MODE', false);

        return parent::install() &&
            $this->registerHook('header') &&
            $this->registerHook('displayAdminOrderMain') && //this is the hook
            $this->registerHook('backOfficeHeader');
    }
```
And then i made a form inside the hook. 
```php
    public function hookDisplayAdminOrderMain(array $params)
    {
        $task = new Task();
        $task->setTask('Write a blog post');
        $task->setDueDate(new \DateTime('tomorrow'));

        $formFactory = $this->get('form.factory');
        $form = $formFactory->createBuilder()
            ->add('task', TextType::class)
            ->add('dueDate', DateType::class)
            ->add('save', SubmitType::class, ['label' => 'Create Task'])
            ->getForm();

        $form->handleRequest($params['request']);

        return $this->render('@Modules/ordererrorreproduce/views/templates/admin/error.html.twig', [
            'form' => $form->createView(),
        ]);
    }
```
Entity of the form:
```php
<?php 
namespace OrderError\Form\Entity;

class Task
{
    protected $task;
    protected $dueDate;

    public function getTask(): string
    {
        return $this->task;
    }

    public function setTask(string $task): void
    {
        $this->task = $task;
    }

    public function getDueDate(): ?\DateTime
    {
        return $this->dueDate;
    }

    public function setDueDate(?\DateTime $dueDate): void
    {
        $this->dueDate = $dueDate;
    }
}
```

the twig:
```twig
{% block content %}
<div class="card">
    <div class="card-body">
        {{ form(form) }}
    </div>
</div>    
{% endblock %}
```

this module will work in 1.7.7.8, but will produce error in 1.7.8. the reason behind it, is that the request is missing from the $params array that the hook provides. Meaning, the issue is that the form here " $form->handleRequest($params['request']);", gets NULL, but in order to handle the form made with form.factory inside the hook, you need that request, but after the update from 1.7.7.8 to 1.7.8 that request object is missing.

-----------------------------------------------------------------------------------------

1. Call the `displayAdminOrderMain` hook in `ps_qualityassurance` module while this hook display its params:
```
public function hookDisplayAdminOrderMain(array $params)
    {
          dump($params);
    }
```

2. BO > SELL > Orders > Edit an Order
3. Check the results below, it display the params
![image](https://user-images.githubusercontent.com/23308427/140385070-ae7c0331-894a-4a2b-b559-177a514f99bf.png)

4. Upgrade your shop to 1.7.8.0
5. BO > SELL > Orders > Edit an Order
6. See error > some params are not displayed
![image](https://user-images.githubusercontent.com/23308427/140385189-3a4ec3ba-8fb1-47b8-95fd-a53edbd17595.png)

**Screenshots**
The error in symfony profiler
![image](https://user-images.githubusercontent.com/23308427/140385283-f4c56be7-b573-4e97-a855-f5b913886de8.png)

#### Additional information

* PrestaShop version: 1.7.8.0
* PHP version: 7.4.24

**Comment by khouloudbelguith** [comment]
Hi @krekken,

Thanks for your issue.
Could you please provide a demo test of using the hooks `displayOrderAdminMain` & `displayOrderAdminSide`.
I used this module https://github.com/PrestaShop/ps_qualityassurance with PS1778.
![image](https://user-images.githubusercontent.com/16067358/139415585-a043fd9c-9e70-4578-ac4b-c8d77f4bc465.png)
![image](https://user-images.githubusercontent.com/16067358/139415608-bce76b8f-f0d7-49ba-9682-2f1e47f4a8cc.png)
![image](https://user-images.githubusercontent.com/16067358/139415659-7901852b-0d0e-4267-a0ef-a06d1c43c6a5.png)

In the BO > Orders > Order details page, I don't have any image displayed.
![image](https://user-images.githubusercontent.com/16067358/139415804-baaf3d14-4d3d-4891-9afc-31c1c02e8f35.png)

Thanks to check and feedback.

**Comment by krekken** [comment]
It appears to be that in 1.7.8 the params array has less data in it than the older version

heres the screenshot of the dump of 1.7.8's param array
![1.7.8](https://i.imgur.com/A9ONcY8.png)
and heres the same thing, but in 1.7.7.8
![1.7.7.8](https://i.imgur.com/ycbUkGi.png)

**Comment by khouloudbelguith** [comment]
@krekken,  I tried to create a custom module to register the new hooks.
[mymodule.zip](https://github.com/PrestaShop/PrestaShop/files/7441680/mymodule.zip)
But on the Order details page, the params are not displayed.
https://watch.screencastify.com/v/WqFIKfz0MbSJoJ0Ezo82
I'm doing something wrong?

Thanks to check and feedback.

**Comment by krekken** [comment]
@khouloudbelguith i did it programmatically in my module's main php file. in the install function i registered the hook

```php
   public function install()
    {
        include(dirname(__FILE__) . '/sql/install.php');
        Configuration::updateValue('ORDERPACKAGES_LIVE_MODE', false);

        return parent::install() &&
            $this->registerHook('header') &&
            $this->registerHook('backOfficeHeader') &&
            $this->registerHook('displayAdminOrderMain');
    }
```
and then i wrote a dump for the params

```php 
    public function hookDisplayAdminOrderMain(array $params)
    {
          dump($params);
    }
```
the hooks worked the same way as it did for you in your video when i used 1.7.7.8, but the error message happened in 1.7.8, not in 1.7.7.8. try it with 1.7.8 to replicate the error message.

**Comment by krekken** [comment]
@khouloudbelguith also i accidentally gave you the wrong hook names. i edited the issue with the correct names

**Comment by khouloudbelguith** [comment]
@krekken, thanks for your feedback.
In fact I tried with PS1788(before the upgrade)
If I Install a module 
```
    public function hookDisplayAdminOrderMain(array $params)
    {
          dump($params);
    }
```

=> there is no display in the BO > Orders > Order details page
But if I install a module 
```
    public function hookDisplayAdminOrderSide(array $params)
    {
	return '<img src="' . Tools::getAdminImageUrl('prestashop-avatar.png') . '" />';
    }


    public function hookDisplayAdminOrderMain(array $params)
    {
       return '<img src="' . Tools::getAdminImageUrl('prestashop-avatar.png') . '" />';
    }

```

I have an image well displayed 
![image](https://user-images.githubusercontent.com/16067358/140067712-d27d6501-737a-4d27-b2c1-c1efd921221a.png)

I attached a screen record
https://watch.screencastify.com/v/SH3OCRB3mpaswDGRllQo

I used two different modules
[mymodule.zip](https://github.com/PrestaShop/PrestaShop/files/7467819/mymodule.zip)
https://github.com/PrestaShop/ps_qualityassurance

Thanks to check and feedback.

**Comment by krekken** [comment]
@khouloudbelguith try using $params['request']. since the modules i'm working with and having issues with display forms in that display hook, i need to handle requests, and for some reason there is no request anymore in the params array. thats why i get the error

**Comment by maldarol** [comment]
I have the same issue if I use the request array in that hook in Prestashop 1.7.8.0.

**Comment by khouloudbelguith** [comment]
> @khouloudbelguith try using $params['request']. since the modules i'm working with and having issues with display forms in that display hook, i need to handle requests, and for some reason there is no request anymore in the params array. thats why i get the error

@krekken, do I need to use this method?

```php
public function hookDisplayAdminOrderMain(array $params)
    {
          dump($params['request']);
    }
```
Can you please confirm?

Thanks!

**Comment by krekken** [comment]
> > @khouloudbelguith try using $params['request']. since the modules i'm working with and having issues with display forms in that display hook, i need to handle requests, and for some reason there is no request anymore in the params array. thats why i get the error
> 
> @krekken, do I need to use this method?
> 
> ```
> public function hookDisplayAdminOrderMain(array $params)
>     {
>           dump($params['request']);
>     }
> ```
> 
> Can you please confirm?
> 
> Thanks!

yes

**Comment by hibatallahAouadni** [comment]
Hello @maldarol 

I just edit your issue description adding the steps to reproduce.
Could you please confirm the steps :pray: 
While calling this hook, did you provide the order_id? 
![image](https://user-images.githubusercontent.com/23308427/140385874-f9eb32da-4b17-4d76-99d2-7c07efec75d9.png)

Please check and feedback.
Thanks!

**Comment by krekken** [comment]
> Hello @maldarol
> 
> I just edit your issue description adding the steps to reproduce. Could you please confirm the steps pray While calling this hook, did you provide the order_id? ![image](https://user-images.githubusercontent.com/23308427/140385874-f9eb32da-4b17-4d76-99d2-7c07efec75d9.png)
> 
> Please check and feedback. Thanks!

it's the other way around, mate. the controller provides the id_order, so you can get it with using the $params array that passes through the data from controller to the module, via the hook.

**Comment by maldarol** [comment]
The problem is indeed not with the hook, as @krekken mentioned the params does not have the request. The error message linked in the original post will appear when you try to render and process a form. Since this class is not a controller you use ``$this->get('form.factory');``, and process the request with `$form->handleRequest($params['request']);`. This is when the given error message will appear.

**Comment by hibatallahAouadni** [comment]
Hello @krekken & @maldarol 

Could you please provide clear and concise steps to reproduce the issue :pray: 
Please check and feedback.

Thanks!

**Comment by krekken** [comment]
> Hello @krekken & @maldarol
> 
> Could you please provide clear and concise steps to reproduce the issue pray Please check and feedback.
> 
> Thanks!

i made a module, that reproduces the problem, but i'll describe what i did, in case you don't want to rely on that. 
https://github.com/krekken/ordererrorreproduce

So i installed the hook like this, in the main php file:

```php

    public function install()
    {
        Configuration::updateValue('ORDERERRORREPRODUCE_LIVE_MODE', false);

        return parent::install() &&
            $this->registerHook('header') &&
            $this->registerHook('displayAdminOrderMain') && //this is the hook
            $this->registerHook('backOfficeHeader');
    }
```
And then i made a form inside the hook. 
```php
    public function hookDisplayAdminOrderMain(array $params)
    {
        $task = new Task();
        $task->setTask('Write a blog post');
        $task->setDueDate(new \DateTime('tomorrow'));

        $formFactory = $this->get('form.factory');
        $form = $formFactory->createBuilder()
            ->add('task', TextType::class)
            ->add('dueDate', DateType::class)
            ->add('save', SubmitType::class, ['label' => 'Create Task'])
            ->getForm();

        $form->handleRequest($params['request']);

        return $this->render('@Modules/ordererrorreproduce/views/templates/admin/error.html.twig', [
            'form' => $form->createView(),
        ]);
    }
```
Entity of the form:
```php
<?php 
namespace OrderError\Form\Entity;

class Task
{
    protected $task;
    protected $dueDate;

    public function getTask(): string
    {
        return $this->task;
    }

    public function setTask(string $task): void
    {
        $this->task = $task;
    }

    public function getDueDate(): ?\DateTime
    {
        return $this->dueDate;
    }

    public function setDueDate(?\DateTime $dueDate): void
    {
        $this->dueDate = $dueDate;
    }
}
```

the twig:
```twig
{% block content %}
<div class="card">
    <div class="card-body">
        {{ form(form) }}
    </div>
</div>    
{% endblock %}
```

this module will work in 1.7.7.8, but will produce error in 1.7.8. the reason behind it, is that the request is missing from the $params array that the hook provides. Meaning, the issue is that the form here " $form->handleRequest($params['request']);", gets NULL, but in order to handle the form made with form.factory inside the hook, you need that request, but after the update from 1.7.7.8 to 1.7.8 that request object is missing.

**Comment by krekken** [comment]
any updates on this?

**Comment by khouloudbelguith** [comment]
Hi  @krekken,

Sorry for the delay.
I manage to reproduce the issue with PS1781 using your module 
![image](https://user-images.githubusercontent.com/16067358/142413017-35ee9091-f0e3-4067-bccf-2bb32918c89f.png)
With PS1778, I have this 
![image](https://user-images.githubusercontent.com/16067358/142413114-5a05b3f8-2186-4e31-85b6-9952b5413880.png)

Following this [comment](https://github.com/PrestaShop/PrestaShop/issues/26471#issuecomment-966883252) 
 ping @PrestaShop/prestashop-maintainers could you please confirm us, it is a bug in the core PrestaShop.
If yes, it is a regression.

Thank you!

**Comment by atomiix** [comment]
Hey @khouloudbelguith I think this is a bug in the core!

**Comment by khouloudbelguith** [comment]
@atomiix, thank you!

**Comment by matks** [comment]
The root cause of https://github.com/PrestaShop/PrestaShop/issues/25248 and this issue is probably the same.

**Comment by Progi1984** [comment]
Fixed in #26699 by @atomiix & tested by @khouloudbelguith.
